cdap-runtime-spi/src/main/java/co/cask/cdap/runtime/spi/profile/Profile.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.runtime.spi.profile;
+
+import co.cask.cdap.runtime.spi.provisioner.ProvisionerSpecification;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Information of a profile. It encapsulates any information required to setup and teardown the program execution
+ * environment. A profile is identified by name and must be assigned a provisioner and its related configuration.
+ */
+public class Profile {
+  private final String name;
+  private final String description;
+  private final ProfileScope scope;
+  private final Map<String, String> primaryProperties;
```

Comment:
this and secondary shouldn't be here. It's just adding a 'required' boolean to the ProvisionerProperty class.